Tasks/AzureAppServiceManage/azureappservicemanage.ts

```diff
@@ -76,91 +33,73 @@ async function run() {
         }
 
-        tl.debug(`Resource Group: ${resourceGroupName}`);    
+        tl.debug(`Resource Group: ${resourceGroupName}`);
+        var appService: AzureAppService = new AzureAppService(azureEndpoint, resourceGroupName, webAppName, slotName);
+        var azureAppServiceUtils: AzureAppServiceUtils = new AzureAppServiceUtils(appService);
+
         switch(action) {
             case "Start Azure App Service": {
-                var appService: AzureAppService = new AzureAppService(azureEndpoint, resourceGroupName, webAppName, slotName);
                 await appService.start();
-                await appService.monitorAppState("running");
-                await appService.pingApplication(pingApplicationCount);
+                await azureAppServiceUtils.monitorApplicationState("running");
+                await azureAppServiceUtils.pingApplication();
                 break;
             }
             case "Stop Azure App Service": {
-                var appService: AzureAppService = new AzureAppService(azureEndpoint, resourceGroupName, webAppName, slotName);
                 await appService.stop();
-                await appService.monitorAppState("stopped");
-                await appService.pingApplication(pingApplicationCount);
+                await azureAppServiceUtils.monitorApplicationState("stopped");
+                await azureAppServiceUtils.pingApplication();
                 break;
             }
             case "Restart Azure App Service": {
-                var appService: AzureAppService = new AzureAppService(azureEndpoint, resourceGroupName, webAppName, slotName);
                 await appService.restart();
                 break;
             }
             case "Swap Slots": {
-                targetSlot = (swapWithProduction) ? productionSlot : targetSlot;
+                targetSlot = (swapWithProduction) ? "production" : targetSlot;
                 var appServiceSourceSlot: AzureAppService = new AzureAppService(azureEndpoint, resourceGroupName, webAppName, sourceSlot);
                 var appServiceTargetSlot: AzureAppService = new AzureAppService(azureEndpoint, resourceGroupName, webAppName, targetSlot);
+                var appServiceSourceSlotUtils: AzureAppServiceUtils = new AzureAppServiceUtils(appServiceSourceSlot);
+                var appServiceTargetSlotUtils: AzureAppServiceUtils = new AzureAppServiceUtils(appServiceTargetSlot);
+
                 if(appServiceSourceSlot.getSlot().toLowerCase() == appServiceTargetSlot.getSlot().toLowerCase()) {
                     updateDeploymentStatus = false;
                     throw new Error(tl.loc('SourceAndTargetSlotCannotBeSame'));
-
                 }
+
                 console.log(tl.loc('WarmingUpSlots'));
-                await appServiceSourceSlot.pingApplication(1);
-                await appServiceTargetSlot.pingApplication(1);
+                await appServiceSourceSlotUtils.pingApplication();
```

Comment:
consider doing ping in parallel and then awaiting both for better perf

Reply:
sure. will do the same.